tests/Functional/Action/Lti/UpdateLti1p3OutcomeActionTest.php

```diff
@@ -0,0 +1,241 @@
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2019 (original work) Open Assessment Technologies S.A.
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\Tests\Functional\Action\Lti;
+
+use OAT\Bundle\Lti1p3Bundle\Tests\Traits\SecurityTestingTrait;
+use OAT\Library\Lti1p3Core\Registration\RegistrationInterface;
+use OAT\Library\Lti1p3Core\Registration\RegistrationRepositoryInterface;
+use OAT\SimpleRoster\Entity\Assignment;
+use OAT\SimpleRoster\Repository\AssignmentRepository;
+use OAT\SimpleRoster\Tests\Traits\DatabaseTestingTrait;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidFactoryInterface;
+use Ramsey\Uuid\UuidInterface;
+use Symfony\Bundle\FrameworkBundle\KernelBrowser;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\HttpFoundation\Response;
+
+class UpdateLti1p3OutcomeActionTest extends WebTestCase
+{
+    use DatabaseTestingTrait;
+    use SecurityTestingTrait;
+
+    /** @var KernelBrowser */
+    private $kernelBrowser;
+
+    /** @var RegistrationInterface */
+    private $registration;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->kernelBrowser = self::createClient();
+        $this->setUpDatabase();
+        $this->loadFixtureByFilename('userWithReadyAssignment.yml');
+
+        /** @phpstan-ignore-next-line */
+        $this->registration = static::$container
+            ->get(RegistrationRepositoryInterface::class)
+            ->find('testRegistration');
+    }
+
+    public function testItReturns401IfNotAuthenticated(): void
+    {
+        $this->kernelBrowser->request('POST', '/api/v1/lti1p3/outcome');
+
+        self::assertEquals(Response::HTTP_UNAUTHORIZED, $this->kernelBrowser->getResponse()->getStatusCode());
+    }
+
+    public function testItReturns200IfTheAuthenticationWorksAndAssignmentExists(): void
+    {
+        $credentials = $this->createTestClientAccessToken(
+            $this->registration,
+            ['https://purl.imsglobal.org/spec/lti-bo/scope/basicoutcome']
+        );
+
+        $uidGenerator = $this->createMock(UuidFactoryInterface::class);
+        self::$container->set('test.uid_generator', $uidGenerator);
+
+        $uidGenerator
+            ->method('uuid4')
+            ->willReturn('e36f227c-2946-11e8-b467-0ed5f89f718b');
+
+        /** @var string $xmlRequestBody */
+        $xmlRequestBody = file_get_contents(
+            __DIR__ . '/../../../Resources/LtiOutcome/valid_replace_result_body.xml'
+        );
+
+        /** @var string $xmlResponseBody */
+        $xmlResponseBody = file_get_contents(
+            __DIR__ . '/../../../Resources/LtiOutcome/valid_replace_result_response_body.xml'
+        );
```

Comment:
Let's try to avoid using such resource files if possible. Since you don't even need a physical file here, why not just having the content as a string? Especially since you expect the UUID you defined above also to be in the response

Reply:
Same answer as above, please fix it